mesonbuild/interpreter.py

```diff
@@ -271,6 +271,10 @@ def __init__(self, pv):
                              'get_unquoted': self.get_unquoted_method,
                              'merge_from': self.merge_from_method,
                              })
+        for name, val in initial_values.items():
+            if not isinstance(val, (int, str)):
+                raise InterpreterException('configuration values must be bool, integer or string')
```

Comment:
With this there are now two different locations where the values are checked. This means they can get out of sync. There really should be only place for verifications.

Could this be changed to work roughly like the following:

 - build empty configuration data object
 - if a dict is given, then call the object's `set_method` for each entry in the dict
 - proceed as usual

Reply:
Good point, PR updated.